frontend/dockerui/namedcontext.go

```diff
@@ -58,15 +59,21 @@ func (bc *Client) namedContext(ctx context.Context, name string, nameWithPlatfor
 
 		named = reference.TagNameOnly(named)
 
-		dgst, data, err := bc.client.ResolveImageConfig(ctx, named.String(), llb.ResolveImageConfigOpt{
+		ref, dgst, data, err := bc.client.ResolveImageConfig(ctx, named.String(), llb.ResolveImageConfigOpt{
 			Platform:     opt.Platform,
 			ResolveMode:  opt.ResolveMode,
 			LogName:      fmt.Sprintf("[context %s] load metadata for %s", nameWithPlatform, ref),
 			ResolverType: llb.ResolverTypeRegistry,
 		})
 		if err != nil {
+			var e imageutil.ResolveToNonImageError
```

Comment:
I think this needs to be a pointer because it looks like `&ResolveToNonImageError{}` is returned when this error happens.

Reply:
^

Reply:
Its passed into `errors.As` as a pointer. We don't ever propagate this error beyond this point, though.

Reply:
https://go.dev/play/p/BQwN9fEeWHl

I don't quite understand how the current one even builds though. I get a build error if I don't remove pointer from `func (e ResolveToNonImageError) Error() string {` in that example.

Reply:
I see the error in the playground but it makes very little sense to me why it would be failing or why the test suite is passing.

Reply:
Oh it is passing because that's a panic not a compile error.

Reply:
It still doesn't look correct to me in the latest version. Now it doesn't error but the `As()` check never matches. It's the case1 vs case2 example in the playground example.

Reply:
Ideally, we could have a test for this as seems to be quite easy to break this behavior.